conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt

```diff
@@ -0,0 +1,1317 @@
+
+    byte _ = result3; // @error smallest K of the operands is 16
+
+    int:Signed8 k = 12;
+    int:Signed16 l = 4434;
+    var result4 = k & l;
+    int:Signed8 _ = result4; // @error the static type of the result is int
+}
+
+Test-Case: output
+Description: Test nil lifting for bitwise and operator.
+Labels: binary-bitwise-expr, byte, DecimalNumber, int, int:Signed8, int:Signed16, int:Signed32, int:Unsigned8, 
+        int:Unsigned16, int:Unsigned32, numeric-nil-lifting, optional-type, unary-minus
+
+function init() {
+    int? a = 43;
+
+    byte? b = 5;
+
+    int:Signed32? c = -21474836;
+    int:Signed16? d = -454;
+    int:Signed8? e = 100;
+
+    int:Unsigned32? f = 21474836;
+    int:Unsigned16? g = 54522;
+    int:Unsigned8? h = 255;
+
+    printResult(a & b); // @output 1
+    printResult(a & c); // @output 40
+    printResult(a & d); // @output 42
+    printResult(a & e); // @output 32
+    printResult(a & f); // @output 0
+    printResult(a & g); // @output 42
+    printResult(a & h); // @output 43
+
+    printResult(b & b); // @output 5
+    printResult(b & c); // @output 4
+    printResult(b & d); // @output 0
+    printResult(b & e); // @output 4
+    printResult(b & f); // @output 4
+    printResult(b & g); // @output 0
+    printResult(b & h); // @output 5
+
+    printResult(c & c); // @output -21474836
+    printResult(c & d); // @output -21474836
+    printResult(c & e); // @output 100
+    printResult(c & f); // @output 4
+    printResult(c & g); // @output 20712
+    printResult(c & h); // @output 236
+
+    printResult(d & d); // @output -454
+    printResult(d & e); // @output 32
+    printResult(d & f); // @output 21474832
+    printResult(d & g); // @output 54330
+    printResult(d & h); // @output 58
+
+    printResult(e & e); // @output 100
+    printResult(e & f); // @output 4
+    printResult(e & g); // @output 96
+    printResult(e & h); // @output 100
+
+    printResult(f & f); // @output 21474836
+    printResult(f & g); // @output 33808
+    printResult(f & h); // @output 20
+
+    printResult(g & g); // @output 54522
+    printResult(g & h); // @output 250
+
+    printResult(h & h); // @output 255
+}
+
+function printResult(int? result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test bitwise and expression evaluates to nil when either operand is nil.
+Labels: binary-bitwise-expr, byte,  DecimalNumber, int, int:Signed8, int:Signed16, int:Signed32, int:Unsigned8, 
+        int:Unsigned16, int:Unsigned32, nil-literal, numeric-nil-lifting, optional-type, unary-minus, value:toBalString
+
+function init() {
+    int? a = 43;
+
+    byte? b = 5;
+
+    int:Signed32? c = -21474836;
+    int:Signed16? d = -454;
+    int:Signed8? e = 100;
+
+    int:Unsigned32? f = 21474836;
+    int:Unsigned16? g = 54522;
+    int:Unsigned8? h = 255;
+
+    int? i = ();
+    byte? j = ();
+    int:Signed32? k = ();
+    int:Unsigned16? l = ();
```

Comment:
usually we don't use letter like `l` for variable names as it is confusing with 1. 
```suggestion
    int:Unsigned16? m = ();
```